app/Http/Controllers/OAuthUserController.php

```diff
@@ -30,6 +30,9 @@ public function callback()
         ]);
 
         Auth::login($user, false);
+        if (Auth::check()) {
```

Comment:
> Technically, I would also get rid of the welcome view, but this is probably an optimization we can do later.

Same comment here, we could avoid these checks if we just decided that the token page is our home page. But that really doesn't matter at this point I think, we can optimize this later if needed.